File: taller-5-java-ronalcastro97/src/main/java/zooAnimales/Ave.java
package zooAnimales;
import gestion.*; 
import java.util.ArrayList;

public class Ave extends Animal{
	private static ArrayList<Ave>listado=new ArrayList<>();
	public static int halcones=0;
	public static int aguilas=0;
	private  String colorPlumas;
	public Ave() {
		super();
		  Ave.listado.add(this);
	}
	public Ave(String nombre, int edad, String habitat, String genero,String colorPlumas) {
		super(nombre, edad, habitat, genero);
		this.colorPlumas=colorPlumas;
		Ave.listado.add(this);
	}
	public static ArrayList<Ave> getListado() {
		return listado;
	}
	public static void setListado(ArrayList<Ave> listado) {
		Ave.listado = listado;
	}
	public static int getHalcones() {
		return halcones;
	}
	public static void setHalcones(int halcones) {
		Ave.halcones = halcones;
	}
	public static int getAguilas() {
		return aguilas;
	}
	public static void setAguilas(int aguilas) {
		Ave.aguilas = aguilas;
	}
	public String getColorPlumas() {
		return colorPlumas;
	}
	public void setColorPlumas(String colorPlumas) {
		this.colorPlumas = colorPlumas;
	}
	@Override
	public String movimiento() {
		return "volar";
	}
	public static Ave crearHalcon(String nombre,int edad,String genero) {
		Ave halcon= new Ave(nombre,edad,"montanas",genero,"cafe glorioso"); 
		Ave.halcones++;
		return halcon 	;	 
	}
	
	public static Ave crearAguila(String nombre,int edad,String genero) {
		Ave aguila=new Ave(nombre,edad,"montanas",genero,"blanco y amarillo");
		Ave.aguilas++;
		return aguila;
	}
	public static int cantidadAves() {
		return Ave.listado.size();
	}
}

File: taller-5-python-ronalcastro97/zooAnimales/ave.py
from zooAnimales.animal import Animal

class Ave(Animal):
    _listado=[]
    halcones=0
    aguilas=0

    def __init__(self, nombre, edad, habitat, genero,colorPlumas):
        super().__init__(nombre, edad, habitat, genero)
        self._colorPlumas=colorPlumas
        Ave._listado.append(self)
    
    def getColorPlumas(self):
        return self._colorPlumas
    
    def setColorPlumas(self,colorPlumas):
        self._colorPlumas=colorPlumas
    
    @classmethod
    def crearHalcon(cls,nombre,edad,genero):
        cls.halcones+=1
        return cls(nombre,edad,"montanas",genero,"cafe glorioso")
    
    @classmethod
    def crearAguila(cls,nombre,edad,genero):
        cls.aguilas+=1
        return cls(nombre,edad,"montanas",genero,"blanco y amarillo")
    
    @staticmethod
    def cantidadAves():
        return len(Ave._listado)
    
    @staticmethod
    def movimiento():
        return "volar"
         
